v2/src/runner.py

```diff
@@ -57,37 +58,75 @@ def outputdir(self) -> pathlib.Path:
         return self.working_directory.joinpath(self.id_.hex)
 
 
-def _create_command(spec: _RunnerSpec) -> str:
-    robot_command = f"{spec.python_executable} -m robot "
-    if spec.variablefile is not None:
-        robot_command += f"--variablefile={spec.variablefile} "
-    if spec.argumentfile is not None:
-        robot_command += f"--argumentfile={spec.argumentfile} "
-    if spec.retry_strategy is _RetryStrategy.INCREMENTAL and spec.previous_output:
-        robot_command += f"--rerunfailed={spec.previous_output} "
-    return robot_command + (
-        f"--outputdir={spec.outputdir} "
-        f"--output={spec.output()} "
-        f"{spec.robot_target}"
-    )
+class _Attempt(Protocol):
+    @property
+    def output(self) -> pathlib.Path:
+        ...
+
+    @property
+    def command(self) -> str:
+        ...
+
+
+@dataclasses.dataclass(frozen=True)
+class _AttemptImpl:
+    runner_spec: _RunnerSpec
+
+    @property
```

Comment:
Oti: Just for the purpose of curiosity: What is the benefit of the @property decorator here? I personally like my function calls to look like function calls.